buildSrc/src/main/java/JniPlugin.java

```diff
@@ -23,34 +21,19 @@
     public void apply(Project project) {
         project.getPluginManager().withPlugin("java", plugin -> {
             JniExtension jniExtension = project.getExtensions().create("jni", JniExtension.class);
-            JniCompilerArguments compilerArguments = new JniCompilerArguments(project.getLayout().getBuildDirectory().dir("generated/jni-headers"));
+            jniExtension.getGeneratedHeadersDirectory().convention(project.getLayout().getBuildDirectory().dir("generated/jni-headers"));
+            JniCompilerArguments compilerArguments = new JniCompilerArguments(jniExtension.getGeneratedHeadersDirectory());
             TaskContainer tasks = project.getTasks();
             TaskProvider<JavaCompile> compileJavaProvider = tasks.named("compileJava", JavaCompile.class);
             RemoveGeneratedNativeHeaders removeGeneratedNativeHeaders = project.getObjects().newInstance(RemoveGeneratedNativeHeaders.class, compilerArguments.getGeneratedHeadersDirectory());
             configureCompileJava(compilerArguments, removeGeneratedNativeHeaders, compileJavaProvider);
-
-            TaskProvider<ConcatenateJniHeaders> concatenateJniHeaders = createConcatenateJniHeadersTask(
+            configureIncludePath(
                 tasks,
-                compileJavaProvider,
-                project.files(compilerArguments.getGeneratedHeadersDirectory()).getAsFileTree(),
-                jniExtension.getGeneratedHeader()
-            );
-
-            configureIncludePath(tasks, concatenateJniHeaders
-                .map(ConcatenateJniHeaders::getGeneratedNativeHeader)
-                .map(it -> it.get().getAsFile().getParentFile())
+                compileJavaProvider.map(compileJava -> compilerArguments.getGeneratedHeadersDirectory())
```

Comment:
I don't understand this mapping here. `compileJava` is not used at all, can we use `compilerArguments.getGeneratedHeadersDirectory()` directly maybe?

Reply:
Yeah, I use this "trick" to add the task dependency to the resulting provider. `compilerArguments.getGeneratedHeadersDirectory()` currently doesn't carry the dependency to the task since it is in a `@Nested` property.

Reply:
I simplified the code, given that the task dependency doesn't work anyway for include directories.